src/ngraph/runtime/cpu/op/as_cast.hpp

```diff
@@ -0,0 +1,181 @@
+#include "ngraph/op/floor.hpp"
+#include "ngraph/op/function_call.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/op/greater.hpp"
+#include "ngraph/op/greater_eq.hpp"
+#include "ngraph/op/less.hpp"
+#include "ngraph/op/less_eq.hpp"
+#include "ngraph/op/log.hpp"
+#include "ngraph/op/lrn.hpp"
+#include "ngraph/op/max.hpp"
+#include "ngraph/op/max_pool.hpp"
+#include "ngraph/op/maximum.hpp"
+#include "ngraph/op/min.hpp"
+#include "ngraph/op/minimum.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/negative.hpp"
+#include "ngraph/op/not.hpp"
+#include "ngraph/op/not_equal.hpp"
+#include "ngraph/op/one_hot.hpp"
+#include "ngraph/op/op.hpp"
+#include "ngraph/op/or.hpp"
+#include "ngraph/op/pad.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/power.hpp"
+#include "ngraph/op/product.hpp"
+#include "ngraph/op/quantize.hpp"
+#include "ngraph/op/reduce.hpp"
+#include "ngraph/op/reduce_window.hpp"
+#include "ngraph/op/relu.hpp"
+#include "ngraph/op/replace_slice.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/op/reverse.hpp"
+#include "ngraph/op/reverse_sequence.hpp"
+#include "ngraph/op/select.hpp"
+#include "ngraph/op/select_and_scatter.hpp"
+#include "ngraph/op/sigmoid.hpp"
+#include "ngraph/op/sign.hpp"
+#include "ngraph/op/sin.hpp"
+#include "ngraph/op/sinh.hpp"
+#include "ngraph/op/slice.hpp"
+#include "ngraph/op/softmax.hpp"
+#include "ngraph/op/sqrt.hpp"
+#include "ngraph/op/stop_gradient.hpp"
+#include "ngraph/op/subtract.hpp"
+#include "ngraph/op/sum.hpp"
+#include "ngraph/op/tan.hpp"
+#include "ngraph/op/tanh.hpp"
+#include "ngraph/op/topk.hpp"
+
+#define NGRAPH_OP(a, b) a,
+enum class OP_TYPEID
+{
+
+#include "ngraph/runtime/cpu/op/op_tbl.hpp"
+
+};
+#undef NGRAPH_OP
+
+template <OP_TYPEID tid>
+struct Hash
+{
+    typedef void* OutputType; // default value, which is dummy
+};
+
+#define NGRAPH_OP(Name, Class)                                                                     \
+    template <>                                                                                    \
+    struct Hash<OP_TYPEID::Name>                                                                   \
+    {                                                                                              \
+        typedef Class::Name OutputType;                                                            \
+    };
+
+#include "ngraph/runtime/cpu/op/op_tbl.hpp"
+
+#undef NGRAPH_OP
+
+static OP_TYPEID get_typeid(const std::string& s)
+{
+#define NGRAPH_OP(a, b) {#a, OP_TYPEID::a},
+    static const std::unordered_map<std::string, OP_TYPEID> typeid_map{
+
+#include "ngraph/runtime/cpu/op/op_tbl.hpp"
+
+    };
+
+#undef NGRAPH_OP
+    auto it = typeid_map.find(s);
+    if (it == typeid_map.end())
+    {
+        throw unsupported_op("Unsupported op '" + s + "'");
+    }
+    return it->second;
+}
+
+template <OP_TYPEID tid>
+static std::shared_ptr<typename Hash<tid>::OutputType> as(std::shared_ptr<ngraph::Node> n)
```

Comment:
You need to document this file. Why was this file created? Why not just use dynamic_cast? The name `as` does not make sense to me but if you documented things perhaps it would.

Reply:
dynamic_cast doesn't work across a shared library boundary on Mac, so any transformer in a shared library can't cast from `Node`.